lib/upload/rever/index.js

```diff
@@ -25,7 +25,8 @@ function buildFormData(args) {
   const payload = buildRequestPayload(args);
 
   if (payload instanceof Blob) {
-    formData.append(FILE_FIELD_NAME, payload, fileName);
+    // Using payload?.name we can preserve the original filename instead of the one with an UUID
+    formData.append(FILE_FIELD_NAME, payload, payload?.name ?? fileName);
```

Comment:
We try to extract the file name from the selected source in web browsers, otherwise, we use the generated name based in a UUID.
It was tested and no conflicts were triggered in backend since we do this only for uploads to rever API (not azure) from web clients.